core/src/main/java/tc/oc/pgm/rotation/MapPoolManager.java

```diff
@@ -152,4 +152,16 @@ public void matchEnded(Match match) {
 
     activeMapPool.matchEnded(match);
   }
+
+  /**
+   * Checks all loaded {@link MapPool}s for a given {@link MapInfo} and returns all {@link MapPool}s
+   * that contains it
+   */
+  public List<MapPool> contains(MapInfo map) {
+    List<MapPool> containedIn = new ArrayList<>();
+    for (MapPool mapPool : this.mapPools) {
+      if (mapPool.getMaps().contains(map)) containedIn.add(mapPool);
+    }
+    return containedIn;
+  }
```

Comment:
This feels so specific that it looks out of place.

Just from wherever you call, do:
```java
poolManager.getPools().stream()
    .filter(p -> p.getMaps().contains(map)).collect(Collectors.toList());
```